### DISCUSSION TITLE
Breeding values relative to other population, say base population

### DISCUSSION THREAD

**Discussion by gregorgorjanc** [gaynorr/AlphaSimR#85] [Ideas]
Breeding value (`bv(pop)`) gives breeding values relative to the current population. Would it be possible to get breeding values relative to some other population, say the base population? The use would be something like

```
bv(pop, base = someOtherPop)
```

**Comment by gaynorr** [comment]
What would you like to use these values for? I ask because you might actually want something slightly different.

**Reply by gregorgorjanc** [reply]
One imminent reason we have for this is that in SIMplyBee we have a single queen per colony and the SIMplyBee colony class has queen, workers, drones, ... slots as AlphaSimR populations. Queen is the only individual and hence her breeding value (compared to her population object) is 0. But in fact, I really should be looking for her breeding value compared to all currently active queens. I could get that by doing the `x = mergePops(...)` and then getting `bv(x)` but though that this type of a problem could be more general hence this question - I bet many animals breeders focused on just breeding values will be asking similar questions sooner or later;)

I am keen to hear about the "slightly" different thing!

**Reply by gregorgorjanc** [reply]
@gaynorr when you said "What would you like to use these values for? I ask because you might actually want something slightly different." did you have in mind the `genParam()` output 

```
...
gv
a matrix of genetic values with dimensions nInd by nTraits

bv
a matrix of breeding values with dimensions nInd by nTraits

dd
a matrix of dominance deviations with dimensions nInd by nTraits

aa
a matrix of additive-by-additive epistatic deviations with dimensions nInd by nTraits

gv_mu
an nTrait vector of intercepts with dimensions nInd by nTraits

gv_a
a matrix of additive genetic values with dimensions nInd by nTraits

gv_d
a matrix of dominance genetic values with dimensions nInd by nTraits

gv_aa
a matrix of additive-by-additive genetic values with dimensions nInd by nTraits
...
```

**Reply by gaynorr** [reply]
It looks like I started to write a reply, got distracted and then forgot that I never finished the reply. Sorry about that.

My thought was that you might be interested in something like breeding values for crossbreds. This would be somewhat similar to calculating breeding values in one population while using another as a reference population. Although, that wouldn't be exactly right because you'd really want gametes from that population as the reference population and not the population itself. If the population is nearly in Hardy-Weinberg equilibrium this distinction wouldn't matter though, but they are not the same if the population isn't in Hardy-Weinberg equilibrium.

For what you do want, it seems that you'd probably still need the queen to sit in a larger population with the other bees. Thus, I don't see how you'd be able to avoid merging populations either way. 

I had considered allowing a reference population be something separate from the current population. For now, I've opted against it because I'm concerned that it might actually introduce more confusion than actually be helpful. The crossbred example above is specifically what I had in mind.

**Reply by gregorgorjanc** [reply]
Thanks for the reply Chris.

Yes, tricky. It feels like that adding a reference/base argument (default would be empty or NULL) to bv() might be the best way to go? By default we take the current population, while a user can provide whatever reference/base population he/she wants. This would cater for queen case, crossbreed case, …, even GCA with her erotic pools?

**Reply by gaynorr** [reply]
It probably makes sense to have a special function for this. GCA would only be correct using this approach if the population was in HWE. In the plant case, that's likely not be the case because you're likely to have inbreds. 

The tricky part is that I already have a function for calculating GCA that does so with the crosses you make. What we want is something that reports this value for all possible crosses. I need to figure out how to clearly distinguish these two options.

**Comment by gregorgorjanc** [comment]
Just linking this related discussion post here https://github.com/gaynorr/AlphaSimR/discussions/106 - take note @janaobsteter

**Reply by gaynorr** [reply]
For GCA, you'd want to follow a slightly different procedure. You'd want to center the population by its allele frequency and not the allele frequency of the other population. You'd also want to use the HWE form of the average effect, if you are assuming the matings are being performed at random.
